packages/amplify-provider-awscloudformation/src/utils/progress-bar-helpers.ts

```diff
@@ -0,0 +1,81 @@
+import columnify from 'columnify';
+
+const COLUMNIFY_WIDTH = 30;
+
+export const CFN_SUCCESS_STATUS = [
+  'UPDATE_COMPLETE', 'CREATE_COMPLETE', 'DELETE_COMPLETE',
+  'DELETE_SKIPPED', 'UPDATE_ROLLBACK_COMPLETE', 'ROLLBACK_COMPLETE',
+];
+export const CNF_ERROR_STATUS = [
+  'CREATE_FAILED', 'DELETE_FAILED', 'UPDATE_FAILED',
+  'UPDATE_ROLLBACK_FAILED', 'ROLLBACK_FAILED',
+];
+
+type ItemPayload = {
+  LogicalResourceId: string,
+  ResourceType: string,
+  ResourceStatus: string,
+  Timestamp: string,
+}
+
+/**
+ * Custom item formatter for progress bar
+ */
+export const createItemFormatter = (payload: ItemPayload): { renderString: string, color: string } => {
+  let color = '';
+  const lowercasePayload = [{
+    logicalResourceId: payload.LogicalResourceId,
+    resourceType: payload.ResourceType,
+    resourceStatus: payload.ResourceStatus,
+    timeStamp: (new Date(payload.Timestamp)).toString(),
+  }];
+
+  const renderString = columnify(lowercasePayload, {
+    showHeaders: false,
+    truncate: true,
+    maxWidth: COLUMNIFY_WIDTH,
+    minWidth: COLUMNIFY_WIDTH,
+  });
+
+  if (CFN_SUCCESS_STATUS.includes(payload.ResourceStatus)) {
+    color = 'green';
+  }
+  if (CNF_ERROR_STATUS.includes(payload.ResourceStatus)) {
+    color = 'red';
+  }
+  return { renderString, color };
+};
+
+type ProgressPayload = {
+  progressName: string,
+  envName: string
+}
+
+/**
+ * Event map type
+ */
+export type EventMap = {
```

Comment:
what's a reason to have another type with same name? do properties differ?
Should both `EventMap` present in this PR get more distinct names ?

Reply:
I agree it's too vague a name. I was going to replace instances of this, but it would cause this PR to balloon in size. I will add this as a follow-up item to refactor the EventMap as it's own PR.